UpToDateStubIndexMismatch: stub count mismatch after "Keywords as identifiers" (missing stub version bump)

## Bug

`com.intellij.psi.stubs.UpToDateStubIndexMismatch` crash when opening files that use V keywords (e.g. `select`) as function/method names.

```
com.intellij.psi.stubs.UpToDateStubIndexMismatch: PSI and index do not match.
...
Caused by: java.lang.AssertionError: Stub count (9) doesn't match stubbed node length (201)
    at com.intellij.psi.impl.source.FileTrees.lambda$reconcilePsi$4(FileTrees.java:188)
```

Triggered by the inspection runner visiting `VlangCircularImportInspection` on files like `vlib/db/sqlite/orm.v`.

## Root Cause

### Primary: missing stub version bump in "Keywords as identifiers" commit

Commit `446520a0` ("Keywords as identifiers", Feb 2026) fixed the parser so V keywords like `select` can appear as function/method names:

```diff
-MethodName ::= identifier | AddOp | MulOp | RelOp
+MethodName ::= <<remapToIdentifier>> identifier | AddOp | MulOp | RelOp
```

However, **the stub version was not bumped**. This leaves stale stub caches for any file that uses a V keyword as a function/method name.

`vlib/db/sqlite/orm.v` is a clear example — its very first declaration is:

```v
pub fn (db DB) select(config orm.SelectConfig, ...) ![][]orm.Primitive { ... }
```

Before the commit, `select` was tokenized as the `SELECT` keyword, causing `MethodName` to fail. The parser's error recovery consumed a large portion of the file, leaving most functions as broken/error nodes. Only ~9 stubs were indexed.

After the commit, `select` parses correctly as a 50-line method with many local variables, type references, etc. The PSI now has 201 stub-eligible nodes. But the cached index still holds 9 stubs (same file timestamp → IntelliJ considers the cache valid). Reconciliation fails with `9 != 201`.

### Secondary: `shouldCreateStubInBlock` uses PSI traversal that is unreliable at reconciliation time

`VlangStubElementType.shouldCreateStub` calls:

```kotlin
protected open fun shouldCreateStubInBlock(node: ASTNode): Boolean {
    return PsiTreeUtil.getParentOfType(node.psi, VlangBlock::class.java) == null
}
```

This traverses the PSI parent chain via `node.psi`. At reconciliation time (inside `PsiFileImpl.loadTreeElement` / `FileTrees.reconcilePsi`), the PSI parent hierarchy may not be fully established for nodes being visited. This causes `getParentOfType` to return `null` even for nodes that are inside a `VlangBlock`, making `shouldCreateStubInBlock` return `true` (create stub) for all nodes — inconsistent with what was actually indexed (where block exclusion worked correctly). This is a latent crash that can occur independently of the version bump issue.

## Fix

1. **Bump `VlangFileElementType.VERSION`** from 95 → 96 to invalidate stale caches and force re-indexing with the corrected parser. This should have been done in the "Keywords as identifiers" commit.

2. **Remove `shouldCreateStubInBlock`** from `VlangStubElementType` so all stub-typed nodes always produce stubs regardless of block context, making indexing and reconciliation consistent.

3. **Override `shouldIndex() = false`** in `VlangVarDefinitionStubElementType` and `VlangLabelDefinitionStubElementType` (make `shouldIndex()` `open` in `VlangNamedStubElementType`) so that local variable and label stubs are not entered into the global `VlangNamesIndex`.

Repeated error from the plugin
```
com.intellij.psi.stubs.UpToDateStubIndexMismatch: PSI and index do not match.
Please report the problem to JetBrains with the files attached
INDEXED VERSION IS THE CURRENT ONE file=V Language file, file.class=class io.vlang.lang.psi.VlangFile, file.lang=Language: vlang, modStamp=1, psi.length=5654
 tree consistent
 stub debugInfo=created in getStubTree(), with AST = false; with backReference
 viewProvider=com.intellij.psi.SingleRootFileViewProvider{vFile=file://S:/repo/vlang/vlib/db/sqlite/orm.v, vFileId=2271, content=VirtualFileContent{size=5654}, eventSystemEnabled=true}
 viewProvider stamp: 0; file stamp: 0; file modCount: 1770123972623; file length: 5654
 doc saved: true; doc stamp: 0; doc size: 5654; committed: true
indexing info: indexing timestamp = 1770123972623, binary = false, byte size = 5654, char size = 5654
ref: 20250127
latestIndexedStub=StubTree{myDebugInfo='created from index; with backReference', myRoot=VlangFileStub}1968757354
   same size=true
   debugInfo=created from index; with backReference
	at com.intellij.psi.stubs.StubTreeLoader.handleUpToDateMismatch(StubTreeLoader.java:224)
	at com.intellij.psi.stubs.StubTreeLoader.access$100(StubTreeLoader.java:25)
	at com.intellij.psi.stubs.StubTreeLoader$StubTreeAndIndexUnmatchCoarseException.doCreateCompleteException(StubTreeLoader.java:216)
	at com.intellij.psi.stubs.StubTreeLoader$StubTreeAndIndexUnmatchCoarseException.lambda$createCompleteException$0(StubTreeLoader.java:192)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:792)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:748)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$4(CoreProgressManager.java:333)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:150)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:333)
	at com.intellij.psi.stubs.StubTreeLoader$StubTreeAndIndexUnmatchCoarseException.createCompleteException(StubTreeLoader.java:191)
	at com.intellij.psi.impl.source.PsiFileImpl.loadTreeElement(PsiFileImpl.java:243)
	at com.intellij.psi.impl.source.PsiFileImpl.calcTreeElement(PsiFileImpl.java:841)
	at com.intellij.psi.impl.source.PsiFileImpl.getChildren(PsiFileImpl.java:846)
	at com.intellij.psi.impl.source.PsiFileImpl.findChildByClass(PsiFileImpl.java:576)
	at io.vlang.lang.psi.VlangFile.getImports(VlangFile.kt:227)
	at io.vlang.ide.inspections.general.VlangCircularImportInspection.buildVisitor$hasCircularImport(VlangCircularImportInspection.kt:22)
	at io.vlang.ide.inspections.general.VlangCircularImportInspection.access$buildVisitor$hasCircularImport(VlangCircularImportInspection.kt:12)
	at io.vlang.ide.inspections.general.VlangCircularImportInspection$buildVisitor$1.visitImportName(VlangCircularImportInspection.kt:41)
	at io.vlang.lang.psi.impl.VlangImportNameImpl.accept(VlangImportNameImpl.java:27)
	at io.vlang.lang.psi.impl.VlangImportNameImpl.accept(VlangImportNameImpl.java:32)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner$InspectionProblemHolder.visitElement(InspectionRunner.java:616)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processContext$19(InspectionRunner.java:416)
	at com.intellij.codeInsight.daemon.impl.InspectionVisitorOptimizer.acceptElements(InspectionVisitorOptimizer.java:208)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.processContext(InspectionRunner.java:414)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$inspect$7(InspectionRunner.java:180)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunReadAction(NestedLocksThreadingSupport.kt:901)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1237)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$executeInImpatientReadAction$16(InspectionRunner.java:393)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:158)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:150)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$executeInImpatientReadAction$17(InspectionRunner.java:384)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:129)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:118)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.executeInImpatientReadAction(InspectionRunner.java:384)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$inspect$8(InspectionRunner.java:173)
	at com.intellij.concurrency.ApplierCompleter.processArrayItem(ApplierCompleter.java:121)
	at com.intellij.concurrency.ApplierCompleter.processArray(ApplierCompleter.java:221)
	at com.intellij.concurrency.ApplierCompleter.helpOthers(ApplierCompleter.java:241)
	at com.intellij.concurrency.ApplierCompleter.lambda$helpAll$3(ApplierCompleter.java:182)
	at com.intellij.concurrency.ThreadContext.resetThreadContext(threadContext.kt:294)
	at com.intellij.concurrency.ApplierCompleter.helpAll(ApplierCompleter.java:181)
	at com.intellij.concurrency.ApplierCompleter.lambda$exec$0(ApplierCompleter.java:111)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunReadAction(NestedLocksThreadingSupport.kt:901)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1237)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$2(ApplierCompleter.java:153)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:717)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:792)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:748)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:716)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:75)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:164)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapAndRun$1(ApplierCompleter.java:145)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:259)
	at com.intellij.concurrency.ApplierCompleter.wrapAndRun(ApplierCompleter.java:145)
	at com.intellij.concurrency.ApplierCompleter.exec(ApplierCompleter.java:113)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.AssertionError: Stub count (9) doesn't match stubbed node length (201)
	at com.intellij.psi.impl.source.FileTrees.lambda$reconcilePsi$4(FileTrees.java:188)
	at com.intellij.psi.impl.DebugUtil.performPsiModification(DebugUtil.java:583)
	at com.intellij.psi.impl.source.FileTrees.reconcilePsi(FileTrees.java:176)
	at com.intellij.psi.impl.source.FileTrees.withAst(FileTrees.java:142)
	at com.intellij.psi.impl.source.PsiFileImpl.loadTreeElement(PsiFileImpl.java:229)
	... 49 more
```

--- a/src/main/kotlin/io/vlang/lang/VlangFileElementType.kt
+++ b/src/main/kotlin/io/vlang/lang/VlangFileElementType.kt
@@ -49,6 +49,6 @@ class VlangFileElementType : IStubFileElementType<VlangFileStub>("VLANG_FILE", V
 
     companion object {
         val INSTANCE = VlangFileElementType()
-        const val VERSION = 95
+        const val VERSION = 96
     }
 }

--- a/src/main/kotlin/io/vlang/lang/stubs/types/VlangLabelDefinitionStubElementType.kt
+++ b/src/main/kotlin/io/vlang/lang/stubs/types/VlangLabelDefinitionStubElementType.kt
@@ -8,6 +8,8 @@ import io.vlang.lang.psi.impl.VlangLabelDefinitionImpl
 import io.vlang.lang.stubs.VlangLabelDefinitionStub
 
 class VlangLabelDefinitionStubElementType(name: String) : VlangNamedStubElementType<VlangLabelDefinitionStub, VlangLabelDefinition>(name) {
+    override fun shouldIndex() = false
+
     override fun createPsi(stub: VlangLabelDefinitionStub): VlangLabelDefinition {
         return VlangLabelDefinitionImpl(stub, this)
     }

--- a/src/main/kotlin/io/vlang/lang/stubs/types/VlangNamedStubElementType.kt
+++ b/src/main/kotlin/io/vlang/lang/stubs/types/VlangNamedStubElementType.kt
@@ -49,7 +49,7 @@ abstract class VlangNamedStubElementType<S : VlangNamedStub<T>, T : VlangNamedEl
         }
     }
 
-    protected fun shouldIndex() = true
+    protected open fun shouldIndex() = true
 
     protected open fun getExtraIndexKeys() = emptyList<StubIndexKey<String, out VlangNamedElement>>()
 }

--- a/src/main/kotlin/io/vlang/lang/stubs/types/VlangStubElementType.kt
+++ b/src/main/kotlin/io/vlang/lang/stubs/types/VlangStubElementType.kt
@@ -1,12 +1,9 @@
 package io.vlang.lang.stubs.types
 
-import com.intellij.lang.ASTNode
 import com.intellij.psi.stubs.IStubElementType
 import com.intellij.psi.stubs.IndexSink
 import com.intellij.psi.stubs.StubBase
-import com.intellij.psi.util.PsiTreeUtil
 import io.vlang.lang.VlangLanguage
-import io.vlang.lang.psi.VlangBlock
 import io.vlang.lang.psi.VlangCompositeElement
 
 abstract class VlangStubElementType<S : StubBase<T>, T : VlangCompositeElement>(debugName: String) :
@@ -15,12 +12,4 @@ abstract class VlangStubElementType<S : StubBase<T>, T : VlangCompositeElement>(
     override fun getExternalId() = "vlang." + super.toString()
 
     override fun indexStub(stub: S, sink: IndexSink) {}
-
-    override fun shouldCreateStub(node: ASTNode): Boolean {
-        return super.shouldCreateStub(node) && shouldCreateStubInBlock(node)
-    }
-
-    protected open fun shouldCreateStubInBlock(node: ASTNode): Boolean {
-        return PsiTreeUtil.getParentOfType(node.psi, VlangBlock::class.java) == null
-    }
 }

--- a/src/main/kotlin/io/vlang/lang/stubs/types/VlangVarDefinitionStubElementType.kt
+++ b/src/main/kotlin/io/vlang/lang/stubs/types/VlangVarDefinitionStubElementType.kt
@@ -8,6 +8,8 @@ import io.vlang.lang.psi.impl.VlangVarDefinitionImpl
 import io.vlang.lang.stubs.VlangVarDefinitionStub
 
 class VlangVarDefinitionStubElementType(name: String) : VlangNamedStubElementType<VlangVarDefinitionStub, VlangVarDefinition>(name) {
+    override fun shouldIndex() = false
+
     override fun createPsi(stub: VlangVarDefinitionStub): VlangVarDefinition {
         return VlangVarDefinitionImpl(stub, this)
     }

